javascript/javascript3/week2/visualpromise/translate-all.js

```diff
@@ -0,0 +1,13 @@
+function translateAllAtOnce() {
+    moveElement(document.querySelector('li:nth-child(1)'), { x: 20, y: 300 })
+    moveElement(document.querySelector('li:nth-child(2)'), { x: 400, y: 300 })
+    moveElement(document.querySelector('li:nth-child(3)'), { x: 400, y: 20 })
+}
+translateAllAtOnce()
+
+const moveElementPromise = moveElement;
+
+Promise.all([moveElementPromise])
+    .then(() => {
+        console.log('All elements have been moved')
+    })
```

Comment:
This will not quite work. Each `moveElement` returns a promise, you need to pass in each promise to `Promise.all` for all it to resolve after all the promises are complete.

Something like:
```
const promise1 = moveElement(document.querySelector('li:nth-child(1)'), { x: 20, y: 300 })
const promise2 = moveElement(document.querySelector('li:nth-child(2)'), { x: 400, y: 300 })
const promise3 = moveElement(document.querySelector('li:nth-child(3)'), { x: 400, y: 20 })

Promise.all([promise1, promise2, promise3])
    .then(() => {
        console.log('All elements have been moved')
    })
```

`promise1`, `promise2`, etc. is generally bad naming (just for the example), a better approach could be:
```
function translateAllAtOnce() {
    return [
        moveElement(document.querySelector('li:nth-child(1)'), { x: 20, y: 300 }),
        moveElement(document.querySelector('li:nth-child(2)'), { x: 400, y: 300 }),
        moveElement(document.querySelector('li:nth-child(3)'), { x: 400, y: 20 }),
    ]
}

Promise.all(translateAllAtOnce())
    .then(() => {
        console.log('All elements have been moved')
    })
```

See i returned the array of promises directly from the function.